cmd/oauth.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/urfave/cli"
+
+	"errors"
+	"github.com/manifoldco/manifold-cli/analytics"
+	"github.com/manifoldco/manifold-cli/api"
+	"github.com/manifoldco/manifold-cli/config"
+	"github.com/manifoldco/manifold-cli/errs"
+	"github.com/manifoldco/manifold-cli/generated/identity/models"
+	"github.com/manifoldco/manifold-cli/prompts"
+	"github.com/manifoldco/manifold-cli/session"
+)
+
+var (
+	linkedOK = "Your Manifold account is now linked"
+)
+
+func init() {
+	oauthCmd := cli.Command{
+		Name:     "oauth",
+		Usage:    "Authenticate with an OAuth provider to login or link accounts",
+		Category: "AUTHENTICATION",
+		Flags: []cli.Flag{
+			cli.BoolFlag{
+				Name:  "github",
+				Usage: "Use the web flow for GitHub authentication",
+			},
+		},
+		Action: oauth,
+	}
+
+	cmds = append(cmds, oauthCmd)
+}
+
+func oauth(cliCtx *cli.Context) error {
+	ctx := context.Background()
+
+	if cliCtx.NumFlags() < 1 {
+		return errs.NewUsageExitError(cliCtx,
+			cli.NewExitError("You must provide an authentication mechanism", -1))
+	}
+
+	cfg, a, err := loadConfigAndAnaltics()
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Unable to load configuration: %s", err), -1)
+	}
+
+	s, err := session.Retrieve(ctx, cfg)
+	if err != nil {
+		cli.NewExitError(fmt.Sprintf("Could not retrieve session: %s", err), -1)
+	}
+
+	if s.Authenticated() {
+		// link
+		ok, err := prompts.Confirm("Do you wish to link your GitHub account to Manifold")
+		if err != nil {
+			if strings.ToLower(ok) == "n" {
+				err = errors.New("user denied prompt")
```

Comment:
I believe Promptui returns a proper error in this case, should we not just return that?